### ISSUE TITLE
`__ziptemp__` not cleaning up

### ISSUE THREAD

**Issue by geoffjukes** [althonos/fs.archive#2]
Hi,

I’m creating a zip using open_fs, inside a temp location. The Zip creates its own, separate `__ziptemp__` location, and doesn’t create the zip until I ‘close’ the zip file. When I close it, the zip is created, but the `__ziptemp__` is not cleaned up. 

I noticed also that all the files from the Zip are in the __ziptemp__. Should I be constantly closing the zip? Or am I doing something wrong?

My use-case is I am copying files from network storage to a local Temp location, renaming them as I do. Sometimes I need the files, sometimes I need them packaged in a zip. So when they need to be zipped, I open_fs a zip:// in the Temp location and copy files there - otherwise I copy the files straight to the Temp://. Once complete, I copy_fs the Temp:// to where it needs to be.

**Comment by geoffjukes** [comment]
Hi,

I refactored to make use of the context handler, but the `__ziptemp__` directory still isn't clearing.

```
            with storage.open('temp://') as temp_fs:
                # Open the source with the modified opener for cifs/nfs rewriting
                source_fs = storage.open(location)
                # Set up a container
                if container:
                    with storage.open_archive(temp_fs, container) as container_fs:
                        for src, tgt in filemap:
                            storage.copy_file(source_fs, src, container_fs, tgt)
                else:
                    for src, tgt in filemap:
                        storage.copy_file(source_fs, src, temp_fs, tgt)
```
The net result is that my workers are filling up their `/tmp` and crashing :(

I'm hoping I don't have to switch back to the starndat ZIP handler in Pyfilesystem2, because I'd like to make use of the ISOs.

**Comment by althonos** [comment]
@geoffjukes : Hi there, thanks for the detailed report. I'll try to see what's going on there, mostly an internal context that's not closed properly I'll try to investigate.

**Comment by geoffjukes** [comment]
Thanks @althonos it's driving me nuts :)

I also tried to re-use the temp_fs that I set up for the overall routine, but the `open_archive` method doesn't support that as an option.

**Comment by geoffjukes** [comment]
@althonos I don't know if it's relevant, but I am using `fs.archive` in an Eventlet enviroment (a Nameko service).
This caused me some issues with `FTPFS`, but I patched `eventlet`s GreenThread to solve that one. (https://github.com/PyFilesystem/pyfilesystem2/issues/141)
My short-term workaround is to run `tmpreaper` on the workers

**Comment by althonos** [comment]
Hi @geoffjukes : as I thought, the temporary filesystem used to fill the archive was not closed. Try the fix I pushed to `master`, if it works I'll publish a new version.

(`pip install --user --upgrade https://github.com/althonos/fs.archive/archive/master.zip`)

**Comment by geoffjukes** [comment]
@althonos It works! Thank you!

And thanks for posting the upgrade instructions too :)

**Comment by althonos** [comment]
Fixed in new `v0.3.1` release.
